controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStoreHelper.java

```diff
@@ -0,0 +1,403 @@
+                    }
+                }, executor);
+    }
+    
+    public CompletableFuture<Void> deleteTable(String scope, String tableName, boolean mustBeEmpty) {
+        log.debug("delete table called for table: {}/{}", scope, tableName);
+        return expectingDataNotFound(withRetries(() -> segmentHelper.deleteTableSegment(
+                scope, tableName, mustBeEmpty, authToken.get(), RequestTag.NON_EXISTENT_ID), 
+                () -> String.format("delete table: %s/%s", scope, tableName)), null)
+                .thenAcceptAsync(v -> log.debug("table {}/{} deleted successfully", scope, tableName), executor);
+    }
+
+    public CompletableFuture<Version> addNewEntry(String scope, String tableName, String key, @NonNull byte[] value) {
+        log.debug("addNewEntry called for : {}/{} key : {}", scope, tableName, key);
+
+        List<TableEntry<byte[], byte[]>> entries = Collections.singletonList(
+                new TableEntryImpl<>(new TableKeyImpl<>(key.getBytes(Charsets.UTF_8), KeyVersion.NOT_EXISTS), value));
+        Supplier<String> errorMessage = () -> String.format("addNewEntry: key: %s table: %s/%s", key, scope, tableName);
+        return withRetries(() -> segmentHelper.updateTableEntries(scope, tableName, entries, authToken.get(), RequestTag.NON_EXISTENT_ID),
+                errorMessage)
+                .exceptionally(e -> {
+                    Throwable unwrap = Exceptions.unwrap(e);
+                    if (unwrap instanceof StoreException.WriteConflictException) {
+                        throw StoreException.create(StoreException.Type.DATA_EXISTS, errorMessage.get());
+                    } else {
+                        log.debug("add new entry {} to {}/{} threw exception {} {}", key, scope, tableName, unwrap.getClass(), unwrap.getMessage());
+                        throw new CompletionException(e);
+                    }
+                })
+                .thenApplyAsync(x -> {
+                    KeyVersion first = x.get(0);
+                    log.debug("entry for key {} added to table {}/{} with version {}", key, scope, tableName, first.getSegmentVersion());
+                    return new Version.LongVersion(first.getSegmentVersion());
+                }, executor);
+    }
+
+    public CompletableFuture<Version> addNewEntryIfAbsent(String scope, String tableName, String key, @NonNull byte[] value) {
+        // if entry exists, we will get write conflict in attempting to create it again. 
+        return expectingDataExists(addNewEntry(scope, tableName, key, value), null);
+    }
+
+    public CompletableFuture<Void> addNewEntriesIfAbsent(String scope, String tableName, Map<String, byte[]> toAdd) {
+        // What happens if we get data exists for one such record? though pravegatablestream, only creates these keys in batch
+        // so ignoring the exception of data exists is safe. But what happens otherwise?
```

Comment:
The comment ends with a question, is this indicating incomplete work, a TODO, or what? I'm not sure what to conclude.

Reply:
I will update the java doc to set the expectations for the caller correctly. 
The thing is - the wire protocol with segment store returns `BadKeyVersion` is any of the keys in the batch already exists. 
So this method also throws the DataExists exception. However, if only a subset of keys existed, it is for the caller to tackle the scenario and identify which keys exist and which dont.

Reply:
How does the caller figure it out? If I have a batch of 100 keys and one exists, how do I determine which one is already there?

Reply:
It cant.. this method is to be used by callers only if they are sure that no one else create keys from the batch otherwise. 

I have updated the javadoc to say that. And that is how we make use of this store helper api.